engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/behavior/OutputCollectionBehaviorTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.bpmn.behavior;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.cloneBuffer;
+import static io.camunda.zeebe.util.buffer.BufferUtil.wrapString;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.el.Expression;
+import io.camunda.zeebe.el.impl.StaticExpression;
+import io.camunda.zeebe.engine.processing.bpmn.BpmnElementContext;
+import io.camunda.zeebe.engine.processing.common.ExpressionProcessor;
+import io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableLoopCharacteristics;
+import io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableMultiInstanceBody;
+import io.camunda.zeebe.msgpack.spec.MsgPackWriter;
+import io.camunda.zeebe.protocol.record.value.ErrorType;
+import io.camunda.zeebe.util.Either;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import java.util.Optional;
+import org.agrona.DirectBuffer;
+import org.agrona.ExpandableArrayBuffer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class OutputCollectionBehaviorTest {
```

Comment:
:x: I'm in favor of removing this unit test. 

Objections:
* too much mocking to verify the logic
* is closely coupled to the implementation
* too many assumptions on how other classes behave
* the behavior is already covered by `MultiInstanceIncidentTest`

I think that the maintenance costs are higher than the benefits. But I'm happy to discuss it f2f.

Reply:
I agree with all your objections. And I think it's worth it. So let's discuss it f2f.

Reply:
I decided to keep it for now. I added a comment in this class, that the future developer is welcome to delete the class if she/he is annoyed by it.